jobs/integration/validation.py

```diff
@@ -2205,32 +2206,82 @@ async def cleanup():
     await cleanup()
 
 
+@dataclass
+class NagiosAlerts:
+    alerts: Mapping[str, bs_ResultSet]
+    matchers: Mapping[str, Callable[[dict], bool]]
+
+    def __bool__(self):
+        """Successful if there are some alerts and all the matchers are True."""
+        if not self.alerts:
+            return False
+        return all(
+            self.matchers[severity](alerts) for severity, alerts in self.alerts.items()
+        )
+
+    def __getitem__(self, item: str) -> bs_ResultSet:
+        return self.alerts[item]
+
+    def __str__(self) -> str:
+        return ", ".join(
+            f"{severity}/{link.string}"
+            for severity, alerts in self.alerts.items()
+            for alert in alerts
+            for link in alert.find_all("a", recursive=False)
+        )
+
+
```

Comment:
this is a `truthy` object that is true when there are some alerts in the mapping and all the matchers are true
if cast to a string it will list all the alerts it has by severity

there's also an accessor method here to `get` the alerts by severity